### ISSUE TITLE
fix(npa/npm): Go marks every method and field public (visibility is detectable from source bytes)

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#458]
## Summary

Go npm and npa count every method/field as public, with a comment claiming "Visibility cannot be detected without source bytes." That comment is **stale** — the `Npm`/`Npa` `compute` signature already receives `code: &[u8]` (other languages use it). Go's export rule is purely lexical (capitalized identifier = exported/public), so unexported members are wrongly counted as public.

## Location

- `src/metrics/npm.rs:595-600` — `G::MethodDeclaration => { class_nm += 1; class_npm += 1; }`
- `src/metrics/npa.rs:972-975` — `class_na += attrs; class_npa += attrs;`
- trait sigs: `src/metrics/npm.rs:247`, `src/metrics/npa.rs:256` (`code: &'a [u8]`)

## Evidence

The Go impls bind the parameter as `_code` and ignore it; other languages (npa.rs:603/716/1793) use `code`. Go visibility is lexical:
```go
type T struct { Name string; secret int }
func (t *T) Greet() {}
func (t *T) helper() {}
```
Measured:
- npm: `class_npm_sum = 2` (both methods) → should be 1; `class_nm_sum = 2` correct.
- npa: `class_npa_sum = 2` → should be 1; `class_na_sum = 2` correct.

This makes Go's COA/CDA accessibility ratios always 1.0.

## Expected Behavior

npm/npa count only exported (capitalized) Go members; unexported (lowercase) members are private.

## Actual Behavior

Every method/field counted as public.

## Impact

npa/npm (and the accessibility ratios derived from them) wrong for every Go file with unexported members. The "cannot detect" comment is factually incorrect and blocks the fix.

---

## Resolution

Fixed in `fe9e7a7f` on `fix/issue-458` (not yet merged).

Go npm/npa now honour Go's lexical export rule: a member is public iff its name's first character is an uppercase Unicode letter (`char::is_uppercase`, so `Ärger` counts). The `Npm`/`Npa` traits already receive `code: &[u8]`; the Go impls now read each member name and gate the public count on it while leaving `nm`/`na` totals unconditional. npa additionally tallies per declared field name (so `A, b int` = 2 attributes) and resolves embedded-field visibility from the embedded type's base name; `_` is never exported. The receiver's visibility does not affect a method's export status.

Stale "cannot detect without source bytes" comments replaced. Revert-verified tests `go_npm_excludes_unexported` / `go_npa_excludes_unexported` added (mixed visibility, multi-name decl, embedded field, blank `_`, Unicode first char), anchored on integer `*_sum` accessors.

Interface methods remain counted as public (out of the `MethodDeclaration`/`FieldDeclaration` scope of this issue).

**Comment by dekobon** [comment]
## Fix Plan

### Root Cause
The Go impls never read the available `code` to check the identifier's first byte.

### Implementation Steps
1. For each `MethodDeclaration` (npm) and field name in `FieldDeclaration` (npa), read the identifier's first byte via `code` and gate the `npm`/`npa` increment on `is_uppercase()`. Keep `nm`/`na` (totals) unconditional.
2. Replace the stale comments.

### Tests to Add
- `go_npm_excludes_unexported`, `go_npa_excludes_unexported` (mixed exported/unexported; revert-verified against all-public).

### Verification
- [ ] `cargo test --workspace --all-features` clean
- [ ] Submodule snapshots accepted/pushed + pointer bump (Go npa/npm shift)
- [ ] Exported-only and mixed structs counted correctly

**Comment by dekobon** [comment]
Fixed in `fe9e7a7f` on `fix/issue-458`.

## What changed
- `src/metrics/npm.rs` — the Go `MethodDeclaration` arm now reads the method-name `field_identifier` from `code` and increments `class_npm` only when `go_is_exported` holds (first char is uppercase Unicode). `class_nm` (total) stays unconditional. The method's own name decides export, not the receiver.
- `src/metrics/npa.rs` — added the shared `go_is_exported` and `go_embedded_field_name` helpers, and rewrote the Go `StructType` field walk to tally **per declared name** with the same export gate. This also fixes a latent undercount: `A, b int` is one `FieldDeclaration` declaring two names, so it now counts 2 attributes (1 public, 1 private) instead of 1. Embedded fields (`io.Reader`, `*Foo`) take visibility from the embedded type's base name; blank `_` is never exported.
- Replaced the stale "cannot detect without source bytes" comments with the lexical-export reality.

## Verification
Measured on the issue's example (`type T struct { Name string; secret int }`, `Greet`/`helper` methods):
- npm: `class_nm_sum = 2`, `class_npm_sum = 1` (was 2).
- npa: `class_na_sum = 2`, `class_npa_sum = 1` (was 2).

Go's COA/CDA ratios are no longer pinned at 1.0.

## Tests (revert-verified)
- `go_npm_excludes_unexported`: exported `Greet`/`Ärger` (Unicode uppercase) + unexported `helper` → nm=3, npm=2. Old code scored npm=3.
- `go_npa_excludes_unexported`: `Name; secret; A, b int; io.Reader; _ int; Ärger bool` → na=7, npa=4. Old code scored na=npa=6.
- Updated `go_struct_fields_are_attributes`, `go_grouped_struct_fields_each_count`, `go_embedded_type_counts_as_attribute` to the corrected per-name + visibility semantics.

Anchored on integer accessors (`class_npm_sum`/`class_nm_sum`/`class_npa_sum`/`class_na_sum`).

## Scope note
Interface methods (`interface { Foo() }`) are left counted as public — they are out of the issue's `MethodDeclaration`/`FieldDeclaration` scope, and unexported interface methods are unusual in Go. Worth a follow-up if desired.

## Validation
`cargo fmt --check`, `cargo clippy --workspace --all-targets --all-features -D warnings`, and `cargo test --workspace --all-features` all pass. No integration `.snap.new` (the corpus has no Go snapshot fixtures). `.bca-baseline.toml` refreshed (headroom) for the slightly larger npa.rs/npm.rs.
